src/lvl_script.c

```diff
@@ -3468,6 +3468,11 @@ void command_computer_dig_to_location(long plr_range_id, const char* origin, con
 
 void command_set_campaign_flag(long plr_range_id, const char *cmpflgname, long val)
 {
+    if (is_map_pack)
+    {
+        SCRPTERRLOG("Command is not applicable to free play levels.");
```

Comment:
This way totally breaks campaign_flags on single levels, that makes it a lot harder to test your stuff. They should work during the level.

Could you change it to a warning? Here and on add_to_campaign_flag both of course.
Then instead of not writing the campaign flags, clear them on level start perhaps?